test/tests/functional/pbs_qstat_dsv.py

```diff
@@ -0,0 +1,259 @@
+# a commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+from ptl.utils.pbs_testsuite import *
+
+
+@tags('smoke')
+class Qstat_dsv(PBSTestSuite):
+    """
+    Test qstat output can be viewed in dsv format
+    """
+
+    def parse_dsv(self, attrs_qstatf, jid, qstat_type, delimiter=None):
+        """
+        Common function to parse qstat output using delimiter
+        """
+        if delimiter:
+            delim = "-D" + str(delimiter)
+        else:
+            delim = " "
+        if qstat_type == "job":
+            cmd = ' -f -F dsv ' + delim + " " + str(jid)
+            qstat_cmd = os.path.join(self.server.pbs_conf['PBS_EXEC'],
+                                     'bin', 'qstat') + cmd
+        elif qstat_type == "server":
+            qstat_cmd = os.path.join(self.server.pbs_conf[
+                'PBS_EXEC'], 'bin', 'qstat') + ' -Bf -F dsv ' + delim
+        elif qstat_type == "queue":
+            qstat_cmd = os.path.join(self.server.pbs_conf[
+                'PBS_EXEC'], 'bin', 'qstat') + ' -Qf -F dsv ' + delim
+        ret = self.du.run_cmd(self.server.hostname, cmd=qstat_cmd, sudo=True)
+        qstat_attrs = []
+        for line in ret['out']:
+            if delimiter:
+                attr_vals = line.split(str(delimiter))
+            else:
+                attr_vals = line.split("|")
+        for item in attr_vals:
+            qstat_attr = item.split("=")
+            qstat_attrs.append(qstat_attr[0])
+        print qstat_attrs
+        rv = True
+        for attr in attrs_qstatf:
+            if attr not in qstat_attrs:
+                print attr + "is missing"
+                rv = False
+                self.assertTrue(rv)
+        self.assertTrue(rv)
+
+    def test_qstat_dsv(self):
+        """
+        test qstat outputs job info in dsv format with default delimiter pipe
+        """
+        j = Job(TEST_USER)
+        j.set_sleep_time(10)
+        jid = self.server.submit(j)
+        print jid
+        qstat_cmd = os.path.join(self.server.pbs_conf[
+            'PBS_EXEC'], 'bin', 'qstat') + ' -f -F dsv ' + str(jid)
+        ret = self.du.run_cmd(self.server.hostname, cmd=qstat_cmd, sudo=True)
+        attrs_qstatf = [
+            'Job_Name', 'Job_Owner', 'job_state', 'queue', 'server',
+            'Checkpoint', 'ctime', 'Error_Path', 'Hold_Types',
+            'Join_Path', 'Keep_Files', 'Mail_Points', 'mtime',
+            'Output_Path', 'Priority', 'qtime', 'Rerunable',
+            'Resource_List.ncpus', 'Resource_List.nodect',
+            'Resource_List.place', 'Resource_List.select', 'schedselect',
+            'substate', 'Variable_List', 'euser', 'egroup', 'queue_rank',
+            'queue_type', 'etime', 'Submit_arguments', 'executable',
+            'argument_list', 'project']
```

Comment:
Thanks for the comment . I will implement this for dsv . For json the format differs , So i will leave the static values as it is.
Json : {
    "timestamp":1492166697,
    "pbs_version":"17.2.0.20170330010525",
    "pbs_server":"localhost",
    "Jobs":{
        "109.localhost":{
  qstat -f : Job Id: 109.localhost
    Job_Name = STDIN